python/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.18...3.18)
+
+set(rmm_version 0.17.0)
+
+project(rmm-python VERSION ${rmm_version} LANGUAGES CXX CUDA)
+
+find_package(Cython REQUIRED)
+find_package(Python3 REQUIRED COMPONENTS Development)
+find_package(CUDAToolkit REQUIRED)
+
+# include(cmake/CPM.cmake)
+# TODO (kk): CPMFindPackage spdlog
+# TODO (kk): CPMFindPackage Thrust
+
+# Try to find an installed RMM, if not found add the parent directory
```

Comment:
If they are building RMM cython from source, wouldn't they be building RMM C++ from source? If so, why look for an installed RMM? You might want to make RMM C++ without installing it (as a developer, I definitely do. I believe rapids-compose works this way).

Reply:
> If they are building RMM cython from source, wouldn't they be building RMM C++ from source? If so, why look for an installed RMM?

Because RMM C++ is header only it's somewhat a moot point, but I wanted to set the right patterns here to repeat elsewhere. Take cudf for example, we'd absolutely want to find a cudf install before calling `add_subdirectory` and rebuilding libcudf. I figured the best bet is to follow suite with RMM.

> You might want to make RMM C++ without installing it (as a developer, I definitely do. I believe rapids-compose works this way).

If an RMM install isn't found it calls an `add_subdirectory` to add the source tree as expected.

Reply:
OK. But for your cuDF example, I would expect to find a *build directory* in cudf/cpp/build* where it is already built, not to `add_subdirectory` and build it again.

Reply:
I suppose you could use `CPMFindPackage` to achieve the same.

Nevertheless, I'm also not quite sure that it's a good idea to look for an installed RMM first, and otherwise build it from source. This certainly brings the danger that one builds the python rmm against an older RMM install lying around somewhere. I think one almost certainly wants to build the python rmm against the included C++ RMM. What one wants to avoid, though, is building RMM twice, once for the C++ install, and then again to build the python against.

In the current way, where there are two distinct builds, I suppose one could build and install C++, and then build python against the just installed version, explicitly passing `RMM_ROOT` to make sure that correct install will be used. In the long run, it might make more sense to unify this into a single cmake build, which will always build the C++ part, and by means of an option additionally builds the python component.

Another minor comment is that it shouldn't be necessary to find, e.g., spdlog here, that should already happen when pulling in the C++ RMM. It reminds me, though, that I need to check whether the new thrust release is out, which if so should allow for finally handling thrust the standard way, too.